thewalrus/_hafnian.py

```diff
@@ -251,3 +280,42 @@ def hafnian_repeated(A, rpt, mu=None, loop=False, rtol=1e-05, atol=1e-08):
         return haf_rpt_complex(A, nud, mu=mu, loop=loop)
 
     return haf_rpt_real(A, nud, mu=mu, loop=loop)
+
+
+def banded_loophaf(A):
+    """Returns the loop hafnian of a banded matrix.
+
+    Args:
+        A (array): a square, symmetric array of even dimensions.
+
+    Returns:
+        np.int64 or np.float64 or np.complex128: the loop hafnian of matrix A.
+    """
+    (n, _) = A.shape
+    w = bandwidth(A)
+    loop_haf = {(): 1, (1,): A[0, 0]}
+    for t in range(1, n + 1):
+        if t - 2 * w - 1 > 0:
+            lower_end = set(range(1, t - 2 * w))
+        else:
+            lower_end = set()
+        upper_end = set(range(1, t + 1))
+        diff = [item for item in upper_end if item not in lower_end]
```

Comment:
maybe there is a more elegant way to this sorted set difference?

Reply:
I think you can do it if you use `frozenset` as type for the key, so that then you have all of the `set` methods.